lib/mini_profiler/profiler.rb

```diff
@@ -553,7 +553,12 @@ def flamegraph(graph)
 
     def ids(env)
       # cap at 10 ids, otherwise there is a chance you can blow the header
-      ([current.page_struct[:id]] + (@storage.get_unviewed_ids(user(env)) || [])[0..8]).uniq
+      all = ([current.page_struct[:id]] + (@storage.get_unviewed_ids(user(env)) || [])).uniq
+      if all.size > 10
+        all = all[0...10]
+        @storage.set_all_unviewed_ids(user(env), all)
```

Comment:
This does not work and has to be renamed to `@storage.set_all_unviewed(user(env), all)`